temporal-testing/src/main/java/io/temporal/internal/testservice/SelfAdvancingTimerImpl.java

```diff
@@ -97,7 +94,13 @@ private void runLocked() {
       while (!Thread.currentThread().isInterrupted()) {
         updateTimeLocked();
         if (!emptyQueue && tasks.isEmpty()) {
-          lockTimeSkippingLocked("runLocked"); // Switching to wall time when no tasks scheduled
+          if (timeLockOnEmptyQueueHandle != null) {
+            throw new IllegalStateException(
+                "SelfAdvancingTimerImpl should have a taken time lock when queue is not empty, but handle is not null");
```

Comment:
Message is confusing here. Is it supposed to say "should *not* have taken" ?

Reply:
Yeah, copy-paste from line 290 with partial editing :) Will fix the message.